src/main/java/com/rtb/projectmanagementtool/user/CreateUserServlet.java

```diff
@@ -24,25 +23,58 @@ public CreateUserServlet() {
   }
 
   // for testing
-  public CreateUserServlet(DatastoreService datastore) {
-    this.datastore = datastore;
-  }
+  // public CreateUserServlet(DatastoreService datastore) {
+  // this.datastore = datastore;
+  // }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // new AuthOps object
     AuthOps auth = new AuthOps(datastore);
 
-    // get Auth ID from AuthOps method
-    String AuthID = auth.getAuthID();
-
     // new UserController object
     UserController controller = new UserController(datastore);
 
-    // create new "blank" UserData object to put in datastore
+    // create new "blank" UserData object
     UserData newUser = controller.createNewUser();
 
-    // set this user's AuthID
+    // new UserService
+    UserService userService = UserServiceFactory.getUserService();
+
+    // if not logged in, redirect to login link
+    if (!userService.isUserLoggedIn()) {
+      // Get login URL
+      String urlToRedirectToAfterUserLogsIn = "/create-new-user";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      // redirect
+      response.sendRedirect(loginUrl);
+    }
+    // if logged in, redirect back to create-new-user
+    else if (userService.isUserLoggedIn()) {
+      response.sendRedirect("/create-new-user.jsp");
+    }
+
+    // get AuthID
+    String AuthID = auth.getAuthID();
```

Comment:
This is the line that has a null pointer.

Reply:
Does this only happen when a user is not logged in?

Reply:
Theoretically the user should be logged in when they reach this line. Line 45 begins the process of login by checking they are not logged in and then redirecting them to a generated loginUrl

Reply:
on your redirects above, you'll probably to do an early return to prevent the rest of the code here from being evaluated.  I'm not sure how instant the response.sedRedirect would be.

Reply:
+1 I noticed that at one point my code had two redirects and the second redirect would run. So it seems like the code continues running even if there's a redirect.